app/vmselect/promql/rollup.go

```diff
@@ -861,13 +863,13 @@ func getScrapeInterval(timestamps []int64, defaultInterval int64) int64 {
 	if scrapeInterval <= 0 {
 		return defaultInterval
 	}
-	return scrapeInterval
+	return applyJitter(scrapeInterval)
 }
 
-func getMaxPrevInterval(scrapeInterval int64) int64 {
-	// Increase scrapeInterval more for smaller scrape intervals in order to hide possible gaps
-	// when high jitter is present.
-	// See https://github.com/VictoriaMetrics/VictoriaMetrics/issues/139 .
+// applyJitter increases scrapeInterval for smaller scrape intervals in order to
+// hide possible gaps when high jitter is present.
+// See https://github.com/VictoriaMetrics/VictoriaMetrics/issues/139
+func applyJitter(scrapeInterval int64) int64 {
```

Comment:
That's really misleading name for the function. It doesn't add jitter to the interval. It adjusts smaller intervals to bigger intervals.

Reply:
Reverted.